src/api/spec/models/project/update_from_xml_command_spec.rb

```diff
@@ -161,9 +161,8 @@
         )
         Project::UpdateFromXmlCommand.new(project).send(:update_repositories, xml_hash, false)
 
-        expect(repository_1.architectures.map(&:name).sort).to eq(['i586', 'x86_64'])
-        expect(repository_1.repository_architectures.where(position: 1).first.architecture.name).to eq('x86_64')
-        expect(repository_1.repository_architectures.where(position: 2).first.architecture.name).to eq('i586')
+        expect(repository_1.architectures.map(&:name)).to eq(['x86_64', 'i586'])
+        expect(repository_1.repository_architectures.map { |repoarch| repoarch.architecture.name }).to eq(['x86_64', 'i586'])
```

Comment:
Just wondering, are the two asserts now testing more or less the same?

Reply:
I would say so - possibly it checks if architectures is sorted as well? But there is a 50:50 chance it's correct anyway :)